### ISSUE TITLE
pass escape key before leaving input

### ISSUE THREAD

**Issue by vogler** [brookhong/Surfingkeys#1407]
Enabling Surfingkeys breaks functionality of input fields that react to the escape key (`esc`) and `onblur`.
I noticed this issue here: https://github.com/chakra-ui/chakra-ui/issues/3198
There, Surfingkeys seems to first unfocus/blur the input, and only after pass `esc`.
First passing `esc` would allow websites to react to the key before `onblur`.

Related issue: https://github.com/brookhong/Surfingkeys/issues/248
Counter-issue: https://github.com/brookhong/Surfingkeys/issues/310 - is this not expected? What about hitting enter before? Is there an alternative to leave input mode without passing escape?

SurfingKeys: 0.9.69
Browser: Chrome 87.0.4280.88 macOS
URL: any input that listens for `esc`, example: https://codesandbox.io/s/happy-hypatia-okuw3?file=/src/App.tsx

**Comment by brookhong** [comment]
Surfingkeys just fires a `blur` on the input, looks like `chakra-ui` will call `onSubmit` on `blur`.

**Comment by vogler** [comment]
Yes, but if Surfingkeys first sent `esc` and then `blur`, there should be no problem with `submitOnBlur` in `chakra-ui` (and others reacting to `esc` and `onblur`).
This order of events makes more sense to me in terms of avoiding conflicts with websites' event handlers.
